torch/distributions/transforms.py

```diff
@@ -231,7 +244,10 @@ def log_abs_det_jacobian(self, x, y):
         result = 0
         for part in self.parts:
             y = part(x)
-            result += part.log_abs_det_jacobian(x, y)
+            term = part.log_abs_det_jacobian(x, y)
+            for _ in range(self.event_dim - part.event_dim):
+                term = term.sum(-1)
```

Comment:
This can be done faster by doing `.contiguous()`, flattening last dimensions and running a single sum.

Reply:
Thanks, I've been looking for a cheap way to do this. I've added a `_sum_rightmost()` helper.